### ISSUE TITLE
[bug] Apple framework paths cached in CMake

### ISSUE THREAD

**Issue by ShadowIce** [conan-io/conan#11218]
### Environment Details (include every applicable attribute)
  * Operating System+version: macOS 12.3.1
  * Compiler+version: clang 13.0.0
  * Conan version: 1.48.0
  * Python version: 3.9.12

### Steps to reproduce (Include if Applicable)

I have a project that depends on a conanized Apple framework. That project uses conan_basic_setup from CMake to setup all conan related things. If I create the CMake cache the first time everything is ok. But if I change the version of the framework and re-run conan+cmake, the CMake cache still references the old version.  I can even delete the conan package from my local cache, re-run conan+cmake and everything seems to work fine until I try to compile it at which point it cannot find the headers for the conan package...

I've analyzed this and found that the cause of this is this call of find_library: https://github.com/conan-io/conan/blob/68378fad044c61a7537fea49c414c2e0f641c9ac/conans/client/generators/cmake_common.py#L859

Solution would be to unset the cache variable first (or use NO_CACHE, but that requires cmake >= 3.21). However I cannot tell whether this is a real bug or intentional behavior. But if it's intentional it behaves different from other libraries I have where I don't need to delete the cmake cache to update the used version.

**Comment by memsharded** [comment]
Hi @ShadowIce 

That integration is deprecated, and it has already been removed in Conan 2.0.
Please use the new integration (``CMakeDeps``, ``CMakeToolchain``, etc), to try it and report. The best starting point is the ``conan new hello/0.1 --template=cmake_lib``, that will generate a working project that you can just ``conan create .`` and test it.

**Comment by ShadowIce** [comment]
I tested it that way, but had to change a few things. First I created an executable (--template=cmake_exe), then added the external framework and ran `conan install . --output-folder cmake-build-release && cd cmake-build-release && cmake .. -DCMAKE_TOOLCHAIN_FILE=build/generators/conan_toolchain.cmake -G Xcode`. This works as expected and the resulting Xcode project has the correct path after changing the version and re-running everything.

I also tried it with `CMakeDeps` but could not figure out how to exactly make that work with cmake. It does show the same bug though. It has its own version of `conan_find_apple_frameworks` which sets a cached variable for `CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND` which never changes once it's set.

**Comment by memsharded** [comment]
I am not sure if this is something we can change, because this is mostly CMake. This variable is defined in:

```cmake
find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAMES ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS} CMAKE_FIND_ROOT_PATH_BOTH)
```

So as you can see, this is a CMake ``find_library()`` call, it is not something Conan is doing. That could mean that the only way CMake has to update to new dependencies like frameworks or libraries path, is to clear the CMake cache?

**Comment by ShadowIce** [comment]
Yes, by default if you use find_library it would cache it. I think this was intended for system libraries that don't change too often, or maybe where the path doesn't include a version string. As soon as you have a lot of libraries that change with nearly every checkout it doesn't make sense (to me at least) to cache anything.

CMake fixed it by introducing the NO_CACHE option for find_library. But that's not used here and it requires version 3.21. The [issue for CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/20687) even mentions conan as an example use case. 😆
